core-plugins/src/main/java/io/cdap/plugin/batch/aggregator/function/Variance.java

```diff
@@ -63,20 +60,26 @@ public void mergeValue(StructuredRecord record) {
       return;
     }
 
-    Number value = (Number) val;
+    Number number = (Number) val;
+    double value = number.doubleValue();
+    double valueSquared = number.doubleValue() * number.doubleValue();
     count++;
 
-    double delta = value.doubleValue() - mean;
+    // Calculate Delta of the value vs the mean and adjust the mean
+    double delta = value - mean;
     mean += delta / count;
-    double delta2 = value.doubleValue() - mean;
-    squareMean += delta * delta2;
+
+    // Calculate delta of the squared value vs the squared mean and adjust the squared mean
+    double deltaSquared = valueSquared - squaredMean;
+    squaredMean += deltaSquared / count;
+
     // if we divide by count, the result will be population variance,
     // count - 1, we will get sample variance
```

Comment:
this comment seems to be unnecessary now

Reply:
You're right. I've updated this comment.